lib/apartment/elevators/first_subdomain.rb

```diff
@@ -10,8 +10,10 @@ module Elevators
     class FirstSubdomain < Subdomain
 
       def parse_tenant_name(request)
-        super.split('.')[0]
+        if super.present?
```

Comment:
This is an [ActiveSupport](http://api.rubyonrails.org/classes/Object.html#method-i-present-3F) core extensions that isn't included by default in non-rails applications (ie Sinatra).
Apartment isn't dependent on Rails itself (+ core ext) it's just dependent on ActiveRecord.

If you actually care about blank strings you can check length, otherwise I'd argue that a simple nil check is fine.